### ISSUE TITLE
Seg-fault in MacOS accessibility bridge

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#80490]
Just upgraded Flutter to ToT (8920f7a5d5), and I'm now frequently seeing the following seg-fault on MacOS:

```
2021-04-14 22:11:17.935 demo[38814:7324858] *** Assertion failure in void flutter::AccessibilityBridgeMacDelegate::DispatchMacOSNotification(__strong gfx::NativeViewAccessible, __strong NSAccessibilityNotificationName)(), ../../flutter/shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm:351
2021-04-14 22:11:17.950 demo[38814:7324858] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The notification target must not be null.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff39659b57 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff7251c5bf objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff39682d08 +[NSException raise:format:arguments:] + 88
	3   Foundation                          0x00007fff3bdb33c6 -[NSAssertionHandler handleFailureInFunction:file:lineNumber:description:] + 166
	4   FlutterMacOS                        0x000000010d48b758 _ZN7flutter30AccessibilityBridgeMacDelegate25DispatchMacOSNotificationEP11objc_objectP8NSString + 600
	5   FlutterMacOS                        0x000000010d4896eb _ZN7flutter30AccessibilityBridgeMacDelegate20OnAccessibilityEventEN2ui16AXEventGenerator13TargetedEventE + 91
	6   FlutterMacOS                        0x000000010dc863ce _ZN7flutter19AccessibilityBridge13CommitUpdatesEv + 2894
	7   FlutterMacOS                        0x000000010d494718 -[FlutterEngine updateSemanticsCustomActions:] + 72
	8   FlutterMacOS                        0x000000010dc9d216 _ZNSt3__110__function6__funcIZ23FlutterEngineInitializeE4$_47NS_9allocatorIS2_EEFvNS_13unordered_mapIiN7flutter25CustomAccessibilityActionENS_4hashIiEENS_8equal_toIiEENS3_INS_4pairIKiS7_EEEEEEEEclEOSG_ + 310
	9   FlutterMacOS                        0x000000010dcae8d6 _ZN7flutter20PlatformViewEmbedder15UpdateSemanticsENSt3__113unordered_mapIiNS_13SemanticsNodeENS1_4hashIiEENS1_8equal_toIiEENS1_9allocatorINS1_4pairIKiS3_EEEEEENS2_IiNS_25CustomAccessibilityActionES5_S7_NS8_INS9_ISA_SE_EEEEEE + 438
	10  FlutterMacOS                        0x000000010da108b8 _ZNSt3__110__function6__funcIZN7flutter5Shell23OnEngineUpdateSemanticsENS_13unordered_mapIiNS2_13SemanticsNodeENS_4hashIiEENS_8equal_toIiEENS_9allocatorINS_4pairIKiS5_EEEEEENS4_IiNS2_25CustomAccessibilityActionES7_S9_NSA_INSB_ISC_SG_EEEEEEE4$_37NSA_ISK_EEFvvEEclEv + 296
	11  FlutterMacOS                        0x000000010dcabde7 _ZN7flutter18EmbedderTaskRunner8PostTaskEy + 759
	12  FlutterMacOS                        0x000000010dc93c9c FlutterEngineRunTask + 44
	13  FlutterMacOS                        0x000000010d494c17 __60-[FlutterEngine postMainThreadTask:targetTimeInNanoseconds:]_block_invoke + 71
	14  libdispatch.dylib                   0x00007fff7366a6c4 _dispatch_call_block_and_release + 12
	15  libdispatch.dylib                   0x00007fff7366b658 _dispatch_client_callout + 8
	16  libdispatch.dylib                   0x00007fff73676cab _dispatch_main_queue_callback_4CF + 936
	17  CoreFoundation                      0x00007fff3961ce81 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	18  CoreFoundation                      0x00007fff395dcc87 __CFRunLoopRun + 2028
	19  CoreFoundation                      0x00007fff395dbe3e CFRunLoopRunSpecific + 462
	20  HIToolbox                           0x00007fff38208abd RunCurrentEventLoopInMode + 292
	21  HIToolbox                           0x00007fff382087d5 ReceiveNextEventCommon + 584
	22  HIToolbox                           0x00007fff38208579 _BlockUntilNextEventMatchingListInModeWithFilter + 64
	23  AppKit                              0x00007fff3684e039 _DPSNextEvent + 883
	24  AppKit                              0x00007fff3684c880 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
	25  AppKit                              0x00007fff3683e58e -[NSApplication run] + 658
	26  AppKit                              0x00007fff36810396 NSApplicationMain + 777
	27  demo                                0x000000010d47473d main + 13
	28  libdyld.dylib                       0x00007fff736c4cc9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Lost connection to device.
```

**Comment by tvolkert** [comment]
I _think_ I was on b921a2443f32e34d0d2c3cfb40009d19141aac44 before upgrading, so it would seem that this was introduced somewhere in between that and 8920f7a5d55af7032ed85056d574d97c97ece13e

**Comment by chunhtai** [comment]
@tvolkert Thanks for reporting this issue. Do you have a repro for this crasher? and does this happen only when the voiceover is turned on?

**Comment by tvolkert** [comment]
I haven't been able to reliably reproduce yet - will update here if I hone in more on when this happens.  But I saw it three times yesterday in 2 hours.

**Comment by tvolkert** [comment]
@chunhtai if there's any extra instrumentation/logging you'd like me to add locally, paste it here, and I can run a local engine and report back the verbose output when it happens again.

**Comment by chunhtai** [comment]
If you can change the line

https://github.com/flutter/engine/blob/a3cdd4994fd13b6c562608ba1dbef3d1a537a40b/shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm#L351

to

```
NSCAssert(native_node, @"The notification target must not be null, the notification is %@", mac_notification);
```

It should give me more info on what is going on.  This only happen when voice over is on, right?

**Comment by tvolkert** [comment]
@chunhtai just got the crash with that modification to the local engine, and the line printed is:

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'The notification target must not be null, the notification is AXCreated'
```

**Comment by chunhtai** [comment]
@tvolkert  thanks for the log, I think i know where the issue is.

**Comment by tvolkert** [comment]
FYI, I can reproduce this 100% of the time now -- by performing a hot restart while VoiceOver is turned on.

**Comment by chunhtai** [comment]
I tried to hot restart, but it didn't crash for me. is there a specific app you  are using?

**Comment by tvolkert** [comment]
Yep - I'm running https://github.com/tvolkert/chicago/tree/master/example at 8920f7a on MacOS

**Comment by tvolkert** [comment]
Steps to repro:

1. Turn on VoiceOver
2. `flutter run -d macos`
3. <R> at the terminal to hot restart
